### PR TITLE
Rewrite text of assignment 4

### PR DESCRIPTION
**Pull request by yogan**

### CODE REVIEW

File: assignment-4.md [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
-* action creators
-* reducer(s) (if you want you can create more than one reducer, but this is up
-  to you)
+We'll start out with the state handling itself, *without integrating it into the
+React app yet*.
 
-Since reducers are always pure functions without side effects it is very
-efficient and easy to unit test them. Write some unit tests that make sure they
-work as intended!
+Create the actions and a reducer for everything that can happen in the app:
```
Comment (RussellSnyder) [inline assignment-4.md:10]:
```suggestion
Create the actions and reducers for everything that can happen in the app:
```

File: assignment-4.md [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
+* reset of minimum/maximum values
+
+Even though the *Redux* code is not yet an active part of our app, we can still
+verify that it works by writing unit tests for it. Remember that *Redux* is
+based on pure functions, which is very convenient in tests: the output is always
+fully defined by the input.
+
+Focus on tests for the *reducer*, which takes in some state and an action, and
+produces a new state as result from that. Think of various scenarios that can
+occur in the app and write them down as test code, asserting the expected new
```
Comment (RussellSnyder) [inline assignment-4.md:23]:
```suggestion
occur in the app and write them down as test code, asserting the new expected```

File: assignment-4.md [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
+state after some action happens with a given starting stating.
```
Comment (RussellSnyder) [inline assignment-4.md:24]: I would rewrite the last line as follows:

Think of various scenarios that can occur in the app and document them as test code. Starting with a mock initial state, assert the new state matches what is expected after an action has been reduced.

It's much more Reduxy ;-)

File: assignment-4.md (hunk 1 above)
Comment (yogan) [reply assignment-4.md:10]: I was thinking about that, and I think it should be just reducer.

In plain Redux, this thing looks like:

```typescript
export function reducer(climate: ClimateState = initialState, action: ClimateActions): ClimateState {
    switch (action.type) {
        case SET_TEMPERATURE:
            return {
                ...climate,
                temperature: _updateTuple(action.temperature, climate.temperature)
            };
        case SET_HUMIDITY:
             // …
        case RESET:
             // …
        default:
             // …
    }
}                                                                                        
```

So it's just a single reducer.

I assume you had the RTK API in mind, where you indeed pass `reduers` to `createSlice`.  
But we're not there yet.

File: assignment-4.md (hunk 3 above)
Comment (yogan) [reply assignment-4.md:24]: Alright, sounds good, but I'll slightly adapt.

Need to drop "mock", as it's not a mock! It's a real state (but not real estate ;-)).

File: assignment-4.md (hunk 1 above)
Comment (seriousManual) [reply assignment-4.md:10]: yeah, I think it would only lead to confusion if we write "reducers"

File: assignment-4.md (hunk 3 above)
Comment (seriousManual) [reply assignment-4.md:24]: badumm tss!

Comment (seriousManual) [reply assignment-4.md:24]: image

**Pull request merged**